### ISSUE TITLE
Custom Azure Policy can't be applied on management groups

### ISSUE THREAD

**Issue by pricoppd** [Azure/azure-policy#256]
Thanks for checking my post.

I'm building custom azure policies using terraform, for some reason all my policies can't be applied on management groups, the policies are only available at subscription level which is not what i need. I tried assigning them from the portal.

I also tried to assign the policy using terraform but they fail with "does not have authorization to perform action 'Microsoft.Authorization/policyAssignments/write' over scope '/providers/Microsoft.Management/ManagementGroups/" although the service principal is having all the necessary permissions Owner, Management Group Contributor, Global Admin.

Can someone explain what i'm missing? i read the documentation but i was not able to figure this out.

**Comment by camillemarie** [comment]
**tl;dr:** If you want to assign a policy to a management group, you must also define it at management group scope.

Policy definitions have a **scope**, which is part of the id, e.g. 

- Policy definition with subscription scope: **/subscriptions/e78961ba-36fe-4739-9212-e3031b4c8db7**/providers/Microsoft.Authorization/policyDefinitions/14bee682-2231-4113-bb3a-c067a49c6035
- Policy definition with management group scope: **/providers/Microsoft.Management/managementGroups/myManagementGroup**/providers/Microsoft.Authorization/policyDefinitions/e1d7de9f-42f0-4af1-9ee0-0187bfce08d5


A policy may only be assigned at a scope "below" the scope where it was defined. 

- If a policy was defined at the scope of a subscription, it may be assigned to that subscription, or resource group within it. It may not be assigned to a management group or any other subscription.
-  If a policy is defined at the scope of a management group, it may be assigned to that management group or any subscription in that management group (or any resource groups within those subscriptions). It may not be assigned to management groups "above" it in the MG hierarchy (aka "ancestors")

I don't know the specifics of how terraform creates policy definitions. If it is creating them at the scope of a subscription by default, you'll need to find a way to create your definitions at management group scope. If this doesn't help, please give an example of what you're doing with terraform. 

It seems like you may also have some permissions issues. Can you see your target management group in the scope selector on the Azure Policy blade in the portal or is it greyed out?

![image](https://user-images.githubusercontent.com/804366/48654032-5b52a100-e9be-11e8-8c46-306be60361c2.png)

**Comment by pricoppd** [comment]
Thank you @camillemarie i already tried what you suggested and it still fails.

azurerm_policy_definition doesn't have a scope attribute, the scope is defined when the policy is applied but by default the policy definition defaults to subscription instead of provider.

i can't find an example to do it in powershell or terraform or azure cli.

**Comment by camillemarie** [comment]
Yes, the command line tools default to the current subscription. However, you can change some parameters so that it gets defined at management group scope.

From [the documentation](https://docs.microsoft.com/en-us/azure/governance/policy/how-to/programmatically-create):

> When called without location parameters, New-AzureRmPolicyDefinition defaults to saving the policy definition in the selected subscription of the sessions context. To save the definition to a different location, use the following parameters:
 SubscriptionId - Save to a different subscription. Requires a GUID value.
ManagementGroupName - Save to a management group. Requires a string value.

See also (scroll to see management group examples):
[Powershell](https://docs.microsoft.com/en-us/powershell/module/azurerm.resources/new-azurermpolicydefinition?view=azurermps-6.12.0#examples)
[Azure CLI](https://docs.microsoft.com/en-us/cli/azure/policy/definition?view=azure-cli-latest#az-policy-definition-create)

**Comment by pilor** [comment]
@pricoppd Terraform doesn't support management group policy defining or assigning yet.  There is an open issue in their repo to add that support:
terraform-providers/terraform-provider-azurerm#2195

**Comment by pilor** [comment]
Assuming this issue was mainly geared towards terraform support since MG policies are supported via the Azure Portal and Azure's first-party programmatic entry points.  If so, the feature ask is covered in the previously mentioned terraform issue.

If that is not the case please re-open this issue.

**Comment by derekschauland** [comment]
Has this changed? When creating a policy definition, all I see are subscriptions for scoping, nowhere can I select management groups (even in the portal)
